tests/hawc/apps/assessment/test_api.py

```diff
@@ -20,47 +20,137 @@ def test_permissions(self, db_keys):
-            (reverse("assessment:api:dataset-data", args=(db_keys.dataset_working,)), 200),
-            (reverse("assessment:api:dataset-version", args=(db_keys.dataset_working, 1)), 403),
-            (reverse("assessment:api:dataset-detail", args=(db_keys.dataset_unpublished,)), 403),
-            (reverse("assessment:api:dataset-data", args=(db_keys.dataset_unpublished,)), 403),
-            (reverse("assessment:api:dataset-version", args=(db_keys.dataset_unpublished, 1)), 403),
+            (
+                reverse("assessment:api:dataset-detail", args=(db_keys.dataset_final,)),
+                200,
+            ),
+            (
+                reverse("assessment:api:dataset-data", args=(db_keys.dataset_final,)),
+                200,
+            ),
+            (
+                reverse("assessment:api:dataset-version", args=(db_keys.dataset_final, 1)),
+                403,
+            ),
+            (
+                reverse("assessment:api:dataset-detail", args=(db_keys.dataset_working,)),
+                200,
+            ),
+            (
+                reverse("assessment:api:dataset-data", args=(db_keys.dataset_working,)),
+                200,
+            ),
+            (
+                reverse("assessment:api:dataset-version", args=(db_keys.dataset_working, 1)),
+                403,
+            ),
+            (
+                reverse("assessment:api:dataset-detail", args=(db_keys.dataset_unpublished,)),
+                403,
+            ),
+            (
+                reverse("assessment:api:dataset-data", args=(db_keys.dataset_unpublished,)),
+                403,
+            ),
+            (
+                reverse(
+                    "assessment:api:dataset-version",
+                    args=(db_keys.dataset_unpublished, 1),
+                ),
+                403,
+            ),
         ]:
             resp = client.get(url)
             assert resp.status_code == code
 
         # unauthenticated can view public final but not private, versions, or unpublished
         client = APIClient()
         for url, code in [
-            (reverse("assessment:api:dataset-detail", args=(db_keys.dataset_final,)), 200),
-            (reverse("assessment:api:dataset-data", args=(db_keys.dataset_final,)), 200),
-            (reverse("assessment:api:dataset-version", args=(db_keys.dataset_final, 1)), 403),
-            (reverse("assessment:api:dataset-detail", args=(db_keys.dataset_working,)), 403),
-            (reverse("assessment:api:dataset-data", args=(db_keys.dataset_working,)), 403),
-            (reverse("assessment:api:dataset-version", args=(db_keys.dataset_working, 1)), 403),
-            (reverse("assessment:api:dataset-detail", args=(db_keys.dataset_unpublished,)), 403),
-            (reverse("assessment:api:dataset-data", args=(db_keys.dataset_unpublished,)), 403),
-            (reverse("assessment:api:dataset-version", args=(db_keys.dataset_unpublished, 1)), 403),
+            (
```

Comment:
@shapiromatron apologies for that. I'm going slightly batty trying to figure out what's different about my ruff environment from the standard. Like when I use "ruff format test_api.py" it is definitely using pyproject.toml for configuration (e.g. if I crank line-length to a very low value and run, it is respecting that)...but these lines for me come out with the spacing:
```
            (
                reverse("assessment:api:dataset-data", args=(db_keys.dataset_final,)),
                200,
            ),
```

instead of the all in one line as you'd expect:
```
            (reverse("assessment:api:dataset-data", args=(db_keys.dataset_final,)), 200),
```

what version of ruff are you (or the CICD environment) using, maybe I'm on something different and that would explain it?

Reply:
we have the ruff version pinned here: https://github.com/shapiromatron/hawc/blob/042492c26b238948c741ed3404ebee57b27ce259/requirements/dev.txt#L8

Reply:
OK @shapiromatron, some of this was me being stupid on a Friday and mixing up the actual github check error (integration test failing as I'd switched the animal version of an assessment being used in those tests) with what was on my mind (formatting issues). That's fixed and this PR now is passing all checks.

Just to check though since you had called this out -- my local ruff (which is running that same 0.1.3 version) and the github check are both happily formatting that line as

```
            (
                reverse("assessment:api:dataset-data", args=(db_keys.dataset_final,)),
                200,
            ),
```

But you mentioned maybe an issue with formatting - are you expecting/seeing something else formatting-wise that I still need to resolve, or is this now good to merge into the new bioassayv2 branch you made? I've made the other suggested changes like renaming subform to formset.